.github/workflows/precommit.yml

```
@@ -0,0 +1,20 @@
+# https://pre-commit.com
+# This GitHub Action assumes that the repo contains a valid .pre-commit-config.yaml file.
+# Using pre-commit.ci is even better that using GitHub Actions for pre-commit.
+name: pre-commit
+on:
+  pull_request:
+    branches: [master]
+  push:
+  workflow_dispatch:
+jobs:
+  pre-commit:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-python@v5
+        with:
+          python-version: 3.x
+      - run: pip install pre-commit
```

To my understanding we could use
```suggestion
      - use: pre-commit/action@v3.0.0
```

Which mostly do the same (pip install and pre-commit run) but also add a cache.

Even if only in maintenance, if the version 3.0.0 had an issue, we could just copy their actions.yaml and fix it here.